### PR TITLE
Replace Faraday with net/http

### PR DESCRIPTION
**Pull request by tsujigiri**

### CODE REVIEW

File: lib/easybill-rest-client/request_builder.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    def body_allowed?
+      HTTP_METHODS_ALLOWING_BODY.include? method
+    end
+
+    def comma_separate_arrays(params)
+      params.map { |k, v| [k, v.is_a?(Array) ? v.join(',') : v] }.to_h
+    end
+
+    def basic_auth_token(api_key)
+      @auth_header ||= "Basic #{Base64.encode64("#{USERNAME}:#{api_key}").delete("\r\n")}"
```
Comment (tsujigiri) [inline lib/easybill-rest-client/request_builder.rb]: `net/http` has [a method for that](http://ruby-doc.org/stdlib-2.3.1/libdoc/net/http/rdoc/Net/HTTPHeader.html#method-i-basic_auth). Do we want to use that instead?

Comment (RKushnir) [inline lib/easybill-rest-client/request_builder.rb]: Ah yeah, I was going to use it but forgot :)

**Comment by tsujigiri** [comment]
:honeybee:

File: lib/easybill-rest-client/request_builder.rb (hunk 1 above)
Comment (RKushnir) [inline lib/easybill-rest-client/request_builder.rb]: Fixed.

**Pull request merged**